### DISCUSSION TITLE
How to access the render buffer/strings while debugging?

### DISCUSSION THREAD

**Discussion by jmbraben** [pallets/jinja#2028] [Q&A]
While using vscode python debugging (with "jinja": true, in launch config) and stepping through a given template, "how" would I see the rendered output as it is being built?
I've tried stepping into the "root_render_func"...but the debugger doesn't support that for some reason (assuming because of how this really works under hood)
"Seeing" what is being rendered would greatly help development
...is there any chance of a vscode plugin that could present the state of the rendering while debugging?
...any other possibilities?

My current workaround is to write output to variables (which are accessible in the debugger) then dump them...but that is not very template friendly

**Comment by mesr** [comment]
Hello there,

I see this is an older post. Is it still current?

I do not know much about vscode, but I have done some good debugging with Jinja2 templates. Can you be more specific about what you'd like to inspect?

If you only want to inspect the contents of the context at a certain point in the rendering, you could simply use the bundled *debug* extension.

If you're trying to inspect the AST resulting from parsing the template, I can share code for a very simple custom Jinja2 extension that does just that.
